src/commands.ts

```diff
@@ -317,7 +309,36 @@ export class CosmosDBCommands {
         }
         return null;
     }
-
+    public static async dropDocDBDatabase(db: DocDBDatabaseNode, explorer: CosmosDBExplorer): Promise<void> {
+        const confirmed = await vscode.window.showWarningMessage("Are you sure to delete database: " + db.label + " and its collections?",
+            "Yes", "No");
+        if (confirmed === "Yes") {
+            const masterKey = await db.getPrimaryMasterKey();
+            const endpoint = await db.getEndpoint();
+            const client = new DocumentClient(endpoint, { masterKey: masterKey });
+            await new Promise((resolve, reject) => {
+                client.deleteDatabase(db.getDbLink(), function (err, result) {
+                    err ? reject(err.body) : resolve(result);
+                });
+            });
+            explorer.refresh(db.server);
+        }
+    }
+    public static async dropDocDBCollection(coll: DocDBCollectionNode, explorer: CosmosDBExplorer): Promise<void> {
+        const confirmed = await vscode.window.showWarningMessage("Are you sure to delete collection: " + coll.label + "?", "Yes", "No");
+        if (confirmed === "Yes") {
+            const masterKey = await coll.db.getPrimaryMasterKey();
+            const endpoint = await coll.db.getEndpoint();
+            const client = new DocumentClient(endpoint, { masterKey: masterKey });
+            const collLink = coll.db.getDbLink() + '/colls/' + coll.id;
```

Comment:
Since you have a 'getDbLink' function, I would prefer you moved this logic into a 'getCollLink' function on the collection node